employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/AbstractPersistable.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain;
+
+import java.io.Serializable;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Version;
+import javax.validation.constraints.NotNull;
+
+import org.optaplanner.core.api.domain.lookup.PlanningId;
+
+@MappedSuperclass
+public abstract class AbstractPersistable implements Serializable {
```

Comment:
May I ask what is the motivation for making the class `Serializable`?

Reply:
This is for persistence. Java objects will be serialized into JSON format to implement CRUD functionality.

Reply:
Does JPA or Hibernate require making the classes serializable? There are very few reasons to use Java serialization nowadays. I found some interesting article putting some light on this topic: https://bvaisakh.wordpress.com/2015/03/04/do-jpa-entities-have-to-be-serializable

Reply:
Hmm, it seems that JPA/Hibernate do not require entities to implement Serializable per se. I didn't know that, good to know! From the article: 

> An entity instance is to be passed by value as a detached object (e.g., through a remote interface), the entity class must implement the Serializable interface.

I don't plan on doing this, so it seems Serializable is not required. I'll fix this in my next commit. Good catch, thanks!